identity/permissions.py

```diff
@@ -0,0 +1,29 @@
+import json
+
+from jwcrypto import jwt
+from rest_framework import permissions
+
+from identity.identity import container
+
+
+class B2BPermission(permissions.BasePermission):
+    """
+    Grants permission if the current user is a superuser.
+    If authentication is successful `request.product` is set
+    to the urn of the product making the request.
+    """
+
+    def has_permission(self, request, view):
+        authorization = request.META.get("HTTP_AUTHORIZATION")
+        if authorization and " " in authorization:
+            auth_type, raw_jwt = authorization.split()
+            if auth_type == "Bearer":
+                try:
+                    validated_jwt = jwt.JWT(key=container.platform_jwks, jwt=raw_jwt)
```

Comment:
we've got no root certificate rotation strategy which is a bit concerning, but I don't think we've got a great way to do rotation. we're probably going to have to just live with this solution until it becomes a problem and we implement a better one

Reply:
I think worst-case for us is rotating the key for platform, then killing all of our pods. Is it a good solution? Probably not, but it definitely would work in less than a few minutes.